Automattic/wp-calypso
# feat(hosting-server-settings): implement MasonryGrid component and use it on server-settings page

Related to https://github.com/Automattic/dotcom-forge/issues/9099

## Proposed Changes
We have white spaces under all cards on Server Settings page. With this PR we are implementing Masonry component which will be used here to fix those white spaces:<br />
<img width="895" alt="Screenshot 2024-09-17 at 12 47 08" src="https://github.com/user-attachments/assets/2ea0f366-f4f4-4198-b117-5995b7eadbf7">

## Testing Instructions
### Test functionality
1. Create Atomic website
2. Open `/sites` -> Select that website -> Open "Server Settings" tab
3. Assert that you see totally the same as it was before, but w/o white spaces under cards<br /><img width="1106" alt="Screenshot 2024-09-17 at 12 49 17" src="https://github.com/user-attachments/assets/5c7192ef-100f-4355-9d37-55904209f941">
4. Expand `What is phpMyAdmin?`
5. Assert that Masonry is recalculated and looks good<br /><img width="522" alt="Screenshot 2024-09-17 at 12 49 37" src="https://github.com/user-attachments/assets/4fa90a5e-50d0-42b6-91e8-88a41a845d3a">
6. Resize browser's window
7. Assert that Masonry is recalculated and looks good

### Test the component
1. Open `http://calypso.localhost:3000/devdocs/design`
2. Scroll to the bottom
3. Assert that you see MassonryGrid example<br /><img width="647" alt="Screenshot 2024-09-17 at 12 58 06" src="https://github.com/user-attachments/assets/810f12ed-cf5c-479e-8da4-da66ca1fa51e">
4. Click on the title
5. Assert that you see example and below the description<br /><img width="1304" alt="Screenshot 2024-09-17 at 12 59 09" src="https://github.com/user-attachments/assets/b064b504-bec1-4035-9030-5f5f3dac1cd8">
